Neo4j.Driver/Neo4j.Driver.IntegrationTests/Examples.cs

```diff
@@ -18,338 +18,571 @@
+                    new Config {ConnectionTimeout = TimeSpan.FromSeconds(15)});
+            }
+            // end::config-connection-timeout[]
+
+            [RequireServerFact]
+            public void TestConfigConnectionTimeoutExample()
+            {
+                // Given
+                using (var driver = CreateDriverWithCustomizedConnectionTimeout(Uri, User, Password))
+                using (var session = driver.Session())
+                {
+                    // When & Then
+                    session.Run("RETURN 1").Single()[0].As<int>().Should().Be(1);
+                }
+            }
         }
 
-        [RequireServerFact]
-        public void Configuration()
+        public class ConfigMaxRetryTimeExample : BaseExample
         {
-            //tag::configuration[]
-            var driver = GraphDatabase.Driver("bolt://localhost:7687", AuthTokens.Basic("neo4j", "neo4j"),
-                Config.Builder.WithMaxIdleSessionPoolSize(10).ToConfig());
-            //end::configuration[]
-            driver.Dispose();
+            public ConfigMaxRetryTimeExample(ITestOutputHelper output, StandAloneIntegrationTestFixture fixture)
+                : base(output, fixture)
+            {
+            }
+
+            // tag::config-max-retry-time[]
+            public IDriver CreateDriverWithCustomizedMaxRetryTime(string uri, string user, string password)
+            {
+                return GraphDatabase.Driver(uri, AuthTokens.Basic(user, password),
+                    new Config {MaxTransactionRetryTime = TimeSpan.FromSeconds(15)});
+            }
+            // end::config-max-retry-time[]
+
+            [RequireServerFact]
+            public void TestConfigMaxRetryTimeExample()
+            {
+                // Given
+                using (var driver = CreateDriverWithCustomizedMaxRetryTime(Uri, User, Password))
+                using (var session = driver.Session())
+                {
+                    // When & Then
+                    session.Run("RETURN 1").Single()[0].As<int>().Should().Be(1);
+                }
+            }
         }
 
-        [RequireServerFact]
-        public void Statement()
+        public class ConfigTrustExample : BaseExample
         {
-            using (var session = Driver.Session())
+            public ConfigTrustExample(ITestOutputHelper output, StandAloneIntegrationTestFixture fixture)
+                : base(output, fixture)
+            {
+            }
+
+            // tag::config-trust[]
+            public IDriver CreateDriverWithCustomizedTrustStrategy(string uri, string user, string password)
+            {
+                return GraphDatabase.Driver(uri, AuthTokens.Basic(user, password),
+                    new Config {TrustStrategy = TrustStrategy.TrustSystemCaSignedCertificates});
+            }
+            // end::config-trust[]
+
+            [RequireServerFact(Skip = "Requires server certificate to be installed on host system.")]
```

Comment:
Maybe we can use `TrustAllCertificates` just for illustration purposes and not skip this test?